### PR TITLE
Authentication token storage

### LINKED ISSUE #172: Store Access Token in Context
**Is your feature request related to a problem? Please describe.**
Access token needs to be stored within the program in order for authentication to work properly. This token must be accessible globally throughout the entire front-end project

**Describe the solution you'd like**
<s>Store the access token in state management.</s>
Store the access token as a cookie.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by saddboys**
Fixes #172 

## Proposed Changes

  - Authentication token now stored as a cookie. To access the cookie, use **document.cookie**, the token is under **threaderAuthToken**. Please note that this may be a temporary storage due to security concerns.

### CODE REVIEW

**Comment by saddboys** [comment]
Suggestion made by @MingChen91 implemented, cookie now stores the token object as a JSON string.

File: frontend/src/pages/Login/Login.jsx [hunk 1]
```diff
@@ -7,10 +7,13 @@ const uiConfig = {
...
     signInSuccessWithAuthResult: (authResult, redirectUrl) => {
-      console.log(authResult); // eslint-disable-line no-console
+      console.log(authResult.credential.idToken); // eslint-disable-line no-console
+      document.cookie = `threaderAuthToken=${JSON.stringify(
+        firebase.auth().currentUser.getIdToken(false),
+      )}`;
       console.log(redirectUrl); // eslint-disable-line no-console
       console.log(firebase.auth().currentUser.getIdToken(false)); // eslint-disable-line no-console
-      return false;
+      return true;
```
Comment (wshin387) [inline frontend/src/pages/Login/Login.jsx:16]: Does this work? Documentation suggests that the 'signInSuccessUrl' is 'Required when the signInSuccessWithAuthResult callback is not used or when it returns true.'

Comment (saddboys) [reply frontend/src/pages/Login/Login.jsx:16]: It doesn't matter whether or not its true or false.

**Comment by saddboys** [comment]
Upon inspection, the JSON object is not storable due to a circular reference.

**Comment by saddboys** [comment]
**DO NOT ACCEPT THIS PR**

File: frontend/src/pages/Login/Login.jsx [hunk 2]
```diff
@@ -7,16 +7,34 @@ const uiConfig = {
...
+      console.log(authResult.credential.idToken); // eslint-disable-line no-console
       console.log(redirectUrl); // eslint-disable-line no-console
-      console.log(firebase.auth().currentUser.getIdToken(false)); // eslint-disable-line no-console
-      return false;
+      document.cookie = `threaderAuthToken=${authResult.user.xa}`;
+
+      return true;
     },
   },
+  signInSuccessUrl: '/',
```
Comment (jacoballen4534) [inline frontend/src/pages/Login/Login.jsx:17]: I don't think you are saving the correct thing? I believe you can just use:
```
  callbacks: {
    signInSuccessWithAuthResult: async () => {
      const threaderAuthToken = await firebase.auth().currentUser.getIdToken(false);
      document.cookie = `threaderAuthToken=${threaderAuthToken}`;
      return true;
    },
  },
```

Comment (saddboys) [reply frontend/src/pages/Login/Login.jsx:17]: I think both those things are the exact same.

Comment (jacoballen4534) [reply frontend/src/pages/Login/Login.jsx:17]: yeah dw, This is referring to before the latest update

File: frontend/src/pages/Login/Login.jsx [hunk 3]
```diff
@@ -7,16 +7,34 @@ const uiConfig = {
...
+
+/**
+ * This Method is for getting the authentication token for the current logged in user from the
+ * cookies within your browser. It returns the Authentication token if it exists, otherwise an empty
+ * string
+ *
+ * @returns {string} which is the auth token
+ * @constructor
+ */
+const GetAuthToken = () => {
```
Comment (Sabatr) [inline frontend/src/pages/Login/Login.jsx:28]: Should be camel case

**Comment by Sabatr** [comment]
LGTM 👍

**Pull request merged**